### PR TITLE
hotfix: XHR abort and surrogate detection

### PR DESCRIPTION
**Pull request by brindy**
Reviewer: Mia or Caine
Asana: 

https://app.asana.com/0/414235014887631/487711692211932
https://app.asana.com/0/414235014887631/550980311876844

CC:

**Description**:

Fixes a couple of broken sites caused by:

* XHRs not aborting (so XHRs would timeout rather than responding immediately)
* Surrogate matching did not take in to account query strings on resource URLs

**Steps to test this PR**:

1. Visit marcus.com - the site should load and render normally (rather than just be a white screen)
2. Visit fillson.com - clicking on one of the large product pictures should navigate to that product
3. Perform exploratory testing as usual

### CODE REVIEW

File: Core/contentblocker.js [hunk 1]
```diff
@@ -218,10 +218,13 @@ var duckduckgoContentBlocking = function() {
 
 	// public 
 	function loadSurrogate(url) {
+		var withoutQueryString = url.split("?")[0]
```
Comment (nilnilnil) [inline Core/contentblocker.js:221]: This makes me a bit nervous. What about all the other (including broken from site owners) permutations?

File: DuckDuckGo.xcodeproj/xcshareddata/xcschemes/AppScreenshots.xcscheme [hunk 2]
```diff
@@ -1,103 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nilnilnil) [inline DuckDuckGo.xcodeproj/xcshareddata/xcschemes/AppScreenshots.xcscheme:1]: Why is this removed?

**Review by nilnilnil — COMMENTED** [review]
Some questions.

File: DuckDuckGo.xcodeproj/xcshareddata/xcschemes/AppScreenshots.xcscheme (hunk 2 above)
Comment (brindy) [reply DuckDuckGo.xcodeproj/xcshareddata/xcschemes/AppScreenshots.xcscheme:1]: We don't generate screenshots via the screenshot now. I already removed the associated code and target but forgot to remove this scheme, so just a clean up.

**Review by nilnilnil — APPROVED** [review]
Still curious about the file removal, but fine on the rest.
